user/user_test.go

```diff
@@ -0,0 +1,50 @@
+package user
+
+import (
+	"math/rand"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestIsValidEmail(t *testing.T) {
```

Comment:
1 test 1 assert or at least 1 expected results. 
you can refactor using table driven tests if you like or split into more tests.
https://github.com/golang/go/wiki/TableDrivenTests